torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -294,6 +296,11 @@ script::Module optimizeForMobile(
     FoldPrePackingOps(cloned_module);
   }
 
+  if (!optimization_blocklist.count(
+      MobileOptimizerType::HOIST_CONV_PACKED_PARAMS)) {
+    HoistConvPackedParams(cloned_module);
```

Comment:
I am assuming this requires inlining? If so then you might want to run freeze_module once again? Since other pass might be disabled?

Reply:
yeah we may need to freeze again to remove the conv module itself I think

Reply:
I also meant before. Doesnt this pass require inlining of the graph as well? Because right now inlining is done by freeze_module, and if you have INSERT_FOLD_PREPACK_OPS blocklisted, it wont do that. So if inlining is required I suggest you call Inline explicitly before calling `HoistConvPackedParams`

Reply:
yeah, I have the same question, we need to make sure what is required before running this pass. I think inline is required in this case. Also actually I'm not sure if it is OK to freeze module for user if it is not frozen previously.

Reply:
@kimishpatel  In general I feel this is an optimization only valid when freeze is enabled, is there a way to enforce that in `optimization_blocklist`?

Reply:
sounds good.  We can throw an assert here for `INSERT_FOLD_PREPACK_OPS` being enabled, unless there is a better way.

Reply:
> @kimishpatel In general I feel this is an optimization only valid when freeze is enabled. why is freezing not an optimization option actually?

@jerryzh168, I think freezing was never thought of as an optimization think when I did this. Your point is fair, maybe we can add that.

@vkuzo, I would say instead of throwing an assert, we just add freezing before this. I understand it may mean we are calling freezing multiple times, but that should not be an issue. Plus that makes this pass self contained without any dependency on having to run another pass.